### PR TITLE
[Bugbash] fix some pruner bugs

### PR DESCRIPTION
**Pull request by J-shang**
### Description ###
Fix some bugs found by @QuanluZhang
1. let pruner raise `unexpected keyword argument` error before raise `required positional argument` error.
2. disable evaluation in functional based scheduled pruners.
3. fix activation-based pruner collected module output on two devices in DP training.
4. fix activation-based pruner always reduce activation dim on 0 (now dynamically adapt reduce dims to different module types)

#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py [hunk 1]
```diff
@@ -747,15 +748,29 @@ def _collector(self, buffer: List) -> Callable[[Module, Tensor, Tensor], None]:
...
+                batch_nums = math.prod(output.shape[:-2])
+                batch_dims = [_ for _ in range(len(output.shape[:-2]))]
+            elif isinstance(_module.module, (torch.nn.Conv2d, torch.nn.ConvTranspose2d)):
+                batch_nums = math.prod(output.shape[:-3])
+                batch_dims = [_ for _ in range(len(output.shape[:-3]))]
+            elif isinstance(_module.module, (torch.nn.Conv3d, torch.nn.ConvTranspose3d)):
+                batch_nums = math.prod(output.shape[:-4])
+                batch_dims = [_ for _ in range(len(output.shape[:-4]))]
+            else:
+                raise TypeError(f'Found unsupported module type in activation based pruner: {_module.__class__.__name__}')
```
Comment (QuanluZhang) [inline nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py:764]: how about put this part into a utility function?

**Review by ultmaster — APPROVED** [review]

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py (hunk 1 above)
Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py]: good suggestion, moved
